specification/source/cdl.rst

```diff
@@ -218,37 +271,255 @@ the
+                             The returned array has the same type as ``s``.
+
+                             Recursive definition:
+                             ``fill(s, n1, n2, n3, ...) = fill( fill(s, n2, n3, ...), n1);``,
+                             ``fill(s,n)={s, s, ..., s}``
+
+                             The function needs two or more arguments; that is ``fill(s)``
+                             is not legal.
+   ========================  ===========================================================
 
-Declaration and Assigning Values to Instances
-.............................................
 
-Instantiation is identical to Modelica.
 
 [For example, to instantiate a gain, one would write
 
 .. code-block:: modelica
 
-   Continuous.Gain myGain(k=-1) "Constant gain of -1" annotation(...);
+   Continuous.Gain gai(k=-1) "Constant gain of -1" annotation(...);
 
 where the documentation string is optional.
-The annotations is typically used
-for the graphical positioning of the instance in a block-diagram.
-]
+The annotation is typically used
+for the graphical positioning of the instance in a block-diagram.]
+
+Using expressions in parameter assignments, and propagating values of parameters
+in a hierarchical formulation of a control sequence, are convenient language constructs
+to express relations between
+parameters. However, most of today's building control product lines do not support
+propagation of parameter values and evaluation of expressions in parameter assignments.
+For CDL to be compatible with this limitation, the ``modelica-json`` translator
+has optional flags, described below, that trigger the evaluation of propagated parameters,
+and that evaluate expressions that involve parameters.
+
+CDL also has a keyword called ``final`` that prevents a declaration to be changed by the user.
+This can be used in a hierarchical controller to ensure that parameter values are propagated to lower level controller
+in such a way that users can only change their value at the top-level location.
+It can also be used in CDL to enforce that different instances of blocks have the same parameter value.
+For example, if a controller samples two signals, then ``final`` could be used to ensure that they
+sample at the same rate.
+However, most of today's building control product lines do not support such a language construct.
+Therefore, while the CDL translator preserves the ``final`` keyword in the ``CDL-JSON`` format,
+a translator from ``CDL-JSON`` to a control product line is allowed to ignore this declaration.
 
-In the assignment of ``parameters``, calculations are allowed.
 
-[For example, a hysteresis block could be configured as follows
+.. note::
+
+   People who implement control sequences that require that values of parameters are identical
+   among multiple instances of blocks
+   should use blocks that take these values as an input, rather than rely on the ``final`` keyword.
```

Comment:
If `final` is used because the integrity of the sequence indeed depends on the parameters propagation then "should" should be replaced by "must" in this sentence no?

Reply:
Thanks, I corrected this.